feat(export): include particle track polylines in VTK exports

w-bonelli opened the pull request:
Addresses #1749, first raised in #1744

- Add `vtkPolyLine` cells to `vtkUnstructuredGrid` exported by `Vtk.write()` to render particle tracks as lines. Previously exported geometry only included points and a `vtkVertex` cell associated with each point (a cell in vtk can be any collection of 1+ points, not identical to modflow's cell as 3D prismatic volume &mdash; a vtk vertex cell has a single point)
- Associate particle track array data (`particleid` and `time`) with points rather than `vtkVertex` cells
- Add notebook demoing particle track export to VTK then load/plot with PyVista
  - add `pyvista` as optional dependency to `pyproject.toml` and `etc/environment.yml`
  - later, a section could be added to demo VTK's Python API, e.g. applying a [`vtkTubeFilter`](https://vtk.org/doc/nightly/html/classvtkTubeFilter.html) to the pathlines with radius proportional to particle mass

jlarsen-usgs reviewed (approved):
@w-bonelli these changes look good and will provide a better way to visualize particle tracks.

w-bonelli commented:
Thanks @jlarsen-usgs. I added a notebook which exports particle tracks to VTK files, then plots/animates them from another VTK-based tool, in this case PyVista since it can generate perspectives similar to Paraview

Also removed the `vtkTubeFilter` exploratory code since you last had a look. That is maybe best left for the user since it is a visualization detail, not really a concern for the particle tracking data itself? (The main purpose of TubeFilter [is to provide variable line thickness with zoom](https://github.com/lorensen/VTKExamples/blob/a195d67ddd662c821431e07c5a14d1804af4153e/src/Python/PolyData/TubeFilter.py#L4).) It could be added to the example notebook in the future though.

mwtoews commented on flopy/export/vtk.py:
```diff
@@ -1260,50 +1265,80 @@ def add_cell_budget(
         Parameters
         ----------
-        points : list
+        points : list or array-like
```
Numpydoc uses [`array_like`](https://numpydoc.readthedocs.io/en/latest/format.html#other-points-to-keep-in-mind) for these types:
```suggestion
        points : list or array_like
```
same for L1276

mwtoews commented on pyproject.toml:
```diff
@@ -72,6 +72,7 @@ optional = [
     "shapely >=1.8",
     "vtk",
     "xmipy",
+    "pyvista",
```
move this up to keep it alphasorted

mwtoews reviewed (approved):
Looks great! Just a few minor nits.

---

Two related future ideas:

- To consider adding a `.to_pyvista()` method to relevant classes (e.g. [here](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/index.html#vtk-pyvista-export)), as this would be cleaner to do than writing a file to be read with PyVista
- Use a [PyVista Jupyter backend](https://docs.pyvista.org/version/stable/user-guide/jupyter/) to make the notebooks more interactive

w-bonelli commented:
Thanks @mwtoews, fixed and also switched to brighter/more consistent pyvista colorscheme

mwtoews commented:
Another way to set the color scheme that uses white in the background is `pv.set_plot_theme("document")` ([docs](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.set_plot_theme.html)).
